pkg/reposync/conditions.go

```diff
@@ -146,20 +158,23 @@ func GetCondition(conditions []v1beta1.RepoSyncCondition, condType v1beta1.RepoS
 }
 
 // removeCondition removes the RepoSync condition with the provided type.
-func removeCondition(rs *v1beta1.RepoSync, condType v1beta1.RepoSyncConditionType) {
-	rs.Status.Conditions = filterOutCondition(rs.Status.Conditions, condType)
+func removeCondition(rs *v1beta1.RepoSync, condType v1beta1.RepoSyncConditionType) (updated bool) {
```

Comment:
Is it intentional that the `updated` return value is unused by the consumers of this function?

Reply:
fixed

Reply:
It still looks like consumers aren't using this

Reply:
SetReconciling & SetStalled use the `transitioned` bool to determine when to remove he Syncing condition. And I need the `updated` condition for the reconciler finalizer PR, where I add SetReconcilerFinalizing, RemoveReconcilerFinalizing, SetReconcilerFinalizerFailure, and RemoveReconcilerFinalizerFailure, all of which return the `updated` bool to decide whether to perform an UPDATE or not. Ideally the reconciler-manager and reconciler would use the `updated` bool for the same purpose, but I don't have time to do that refactor right now. So I just exposed both bools on all the Setter functions so we can use them later.

Reply:
Sounds good